tests/phpunit/CRM/Contact/Form/Task/PDFLetterCommonTest.php

```diff
@@ -153,4 +155,57 @@ public function testContactTokensAreResolved(): void {
     $this->assertStringContainsString('Logged In, Dear Logged In', $processedMessage);
   }
 
+  /**
+   * Test case tokens are resolved in pdf letter.
+   */
+  public function testCaseTokensAreResolved() : void {
+    // @todo - find a better way to set case id....
+    $_REQUEST['caseid'] = $this->getCaseID();
+    $form = $this->getPDFForm([
+      'html_message' => '{contact.first_name}, {case.case_type_id:label} {case.' . $this->getCustomFieldName('text') . '}',
+    ], [$this->contactId]);
+    $processedMessage = $this->submitForm($form)['html'];
+    $this->assertStringContainsString('Logged In, Housing Support', $processedMessage);
```

Comment:
Suggest adding "bb" to the end of the expected string here.

But otherwise PR looks ok.